src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs

```diff
@@ -166,6 +172,10 @@ public void Connect(IPEndPoint remoteEP)
 
         public void Connect(IPAddress[] ipAddresses, int port)
         {
+            if (OperatingSystem.IsWasi()) throw new PlatformNotSupportedException(); // TODO remove with https://github.com/dotnet/runtime/pull/107185
```

Comment:
As @dicej pointed out this means we won't be able to use the sqlclient without some follow ups since it uses sync (on async).

Reply:
correct.